javascript/code-challenges/challenges-06.test.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named getNames that, given an array of people objects, uses map to return an array of names reversed.
+
+For example:
+[
+{
+  name: 'lloyd',
+  age: 32,
+  shoeSize: 12
+},
+{
+  name: 'jamie',
+  age: 21,
+  shoeSize: 8
+}
+]
+
+Returns: ['dyoll', 'eimaj'];
+------------------------------------------------------------------------------------------------ */
+
+const getNames = (arr) => {
+  arr.map((element) => {
+    let str = arr.name;
+    return str.split("").reverse().join("");
+  });
```

Comment:
You're really close to a working solution. Just play with what you have a little more.
I believe line 27 is what's causing your issue.  Think also about how map works. Currently you're mapping through each piece of data within the array and you've named those variable "element" but then you're not doing anything to them. Remember, your input, **arr** is not itself an object. It is an array of objects, so **arr.name** doesn't exist. Each piece of data(element) within your array, however, is an object. So **element.name** would get you access to the property you want.